mautrix_facebook/portal.py

```diff
@@ -575,6 +586,15 @@ def _get_facebook_reply(self, reply: str) -> Optional[RelatesTo]:
             content = await self._convert_facebook_location(intent, attachment)
             content.relates_to = self._get_facebook_reply(reply_to)
             event_id = await intent.send_message(self.mxid, content)
+        elif isinstance(attachment, ShareAttachment):
+            # Prevent sending urls that are already in the original message text
+            if attachment.original_url not in message_urls:
+                content = TextMessageEventContent(msgtype=MessageType.TEXT, body=f"{attachment.title}: {attachment.original_url}")
+                content.format = Format.HTML
+                content.formatted_body = f"<a href='{attachment.original_url}'>{attachment.title}</a>"
+                event_id = await intent.send_message(self.mxid, content)
```

Comment:
Sending these as separate messages seems kind of wasteful, although I guess it's fine temporarily.